### DISCUSSION TITLE
Consume webservices (rest/soap)

### DISCUSSION THREAD

**Discussion by mayermatheus** [fastapi/fastapi#7650] [Questions] [answered]
### First check

* [ x ] I used the GitHub search to find a similar issue and didn't find it.
* [ x ] I searched the FastAPI documentation, with the integrated search.
* [ x ] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I consume a webservice written in restfull / soap?

#### Project scenario:
I will build a robust api with several features involving workflow as well as payment flow. With this, in the project I will have to consume several web services (soap / rest) to make integrations with other systems.

**Comment by zero-shubham** [comment]
isn't integrating/using httpx in for consuming rest apis an option? I'm not sure but I would have done it that way.

**Comment by jorgerpo** [comment, marked as answer]
use httpx

https://news.ycombinator.com/item?id=22012215

**Comment by mayermatheus** [comment]
Thanks @jorgerpo @zero-shubham

**Comment by tiangolo** [comment]
Thanks for the help here @zero-shubham and @jorgerpo !

Yes, HTTPX would be the best match for FastAPI.

If that solves your problem, may we close this issue @mayermatheus ?

**Comment by mayermatheus** [comment]
Yes, @tiangolo
